src/test/java/seedu/address/storage/ReminderPersonsTest.java

```diff
@@ -0,0 +1,170 @@
+
+        // creating a standalone HashMap
+        HashMap<Person, Reminder> personReminderHashMap = new HashMap<>();
+
+        // adding Reminders for model Persons
+        Person alice = new PersonBuilder().withName("Alice").build();
+        Reminder reminderForAlice = new Reminder("arrange home viewing");
+        reminderPersons.add(alice, reminderForAlice);
+
+        Person bob = new PersonBuilder().withName("Bob").build();
+        Reminder reminderForBob = new Reminder("update client's phone number");
+        reminderPersons.add(bob, reminderForBob);
+
+        // adding persons & their reminders to the HashMap
+        personReminderHashMap.put(alice, reminderForAlice);
+        personReminderHashMap.put(bob, reminderForBob);
+
+        // check if ReminderPersons & the standalone HashMap return the same KeySet
+        assertEquals(personReminderHashMap.keySet(), reminderPersons.getKeySet());
+
+        // check if the ReminderPersons object contains the specified Persons added earlier
+        assertTrue(reminderPersons.containsKey(alice));
+        assertTrue(reminderPersons.containsKey(bob));
+
+        // check if the HashMap in the ReminderPersons object and standalone HashMap are the same
+        HashMap<Person, Reminder> reminderPersonsHashMapClone = reminderPersons.clone();
+        assertTrue(reminderPersonsHashMapClone.equals(personReminderHashMap));
+    }
+
+    @Test
+    void isEmpty_success() {
+        // creating a plain copy of ReminderPersons for testing purposes
+        ReminderPersons reminderPersons = ReminderPersons.getInstance();
+        reminderPersons.setTestingInstance();
+
+        // check if the HashMap in reminderPersons is empty
+        assertTrue(reminderPersons.isEmpty());
+
+        // add Reminders for model Persons
+        Person alice = new PersonBuilder().withName("Alice").build();
+        Reminder reminderForAlice = new Reminder("arrange home viewing");
+        reminderPersons.add(alice, reminderForAlice);
+
+        Person bob = new PersonBuilder().withName("Bob").build();
+        Reminder reminderForBob = new Reminder("update client's phone number");
+        reminderPersons.add(bob, reminderForBob);
+
+        // check if the HashMap in reminderPersons is not empty
+        assertFalse(reminderPersons.isEmpty());
+    }
+
+    @Test
+    void toggleFavouriteForReminder_success() {
+        // creating a plain copy of ReminderPersons for testing purposes
+        ReminderPersons reminderPersons = ReminderPersons.getInstance();
+        reminderPersons.setTestingInstance();
+
+        // add Reminder for model Person
+        Person alice = new PersonBuilder().withName("Alice").build();
+        Reminder reminderForAlice = new Reminder("arrange home viewing");
+        reminderPersons.add(alice, reminderForAlice);
+        ReminderPersons.toggleFavouriteForReminder(alice);
+
+        // favoriting a Person
+        Person aliceFavorited = alice.toggleFavourite();
```

Comment:
Just a small nit: Favourite rather than favorite

Reply:
thanks for pointing that out! will fix it in another PR